### PR TITLE
Add `--all-packages` option in remove command

### PR DESCRIPTION
**Pull request by pixyzehn**
- Add `--all-packages` to remove all packages
- Fix my typo🙇 ( `"To remove all packages, pass the '--all-script-data' flag"` )

You can remove all script data and all packages at the same time like this.

```sh
marathon remove --all-script-data --all-packages
🗑  Removed all script data and all packages
```

### CODE REVIEW

File: Sources/MarathonCore/PackageManager.swift [hunk 1]
```diff
@@ -176,6 +176,12 @@ internal final class PackageManager {
         return package
     }
 
+    func removeAllPackages() throws {
+        for package in addedPackages {
+            _ = try removePackage(named: package.name)
```
Comment (JohnSundell) [inline Sources/MarathonCore/PackageManager.swift]: Can you annotate `removePackage(named:)` with `@discardableResult` instead, so we don't have to do the `_`?

File: Sources/MarathonCore/Remove.swift [hunk 2]
```diff
@@ -36,9 +36,20 @@ internal final class RemoveTask: Task, Executable {
...
+                try scriptManager.removeAllScriptData()
+                deletedObjects.append("all script data")
+            }
+
+            if arguments.contains("--all-packages") {
+                try packageManager.removeAllPackages()
+                deletedObjects.append("all packages")
+            }
+
+            return "🗑  Removed \(deletedObjects.joined(separator: " and "))"
```
Comment (JohnSundell) [inline Sources/MarathonCore/Remove.swift]: Clever solution! 👍

File: Tests/MarathonTests/MarathonTests.swift [hunk 3]
```diff
@@ -88,6 +88,22 @@ class MarathonTests: XCTestCase {
...
+        try run(with: ["add", "git@github.com:JohnSundell/Files.git"])
+        try run(with: ["add", "git@github.com:JohnSundell/Wrap.git"])
+        try run(with: ["add", "git@github.com:JohnSundell/Unbox.git"])
+
+        let generatedFolder = try folder.subfolder(atPath: "Packages/Generated")
+        let packagesFolder = try generatedFolder.subfolder(named: "Packages")
+        XCTAssertEqual(packagesFolder.subfolders.count, 3)
+        try XCTAssertEqual(folder.subfolder(named: "Packages").files.count, 3)
+
+        // Remove tht all packages
```
Comment (JohnSundell) [inline Tests/MarathonTests/MarathonTests.swift]: The `tht` should be removed here right? :)

File: Tests/MarathonTests/MarathonTests.swift [hunk 4]
```diff
@@ -88,6 +88,22 @@ class MarathonTests: XCTestCase {
         try XCTAssertFalse(run(with: ["list"]).contains(packageFolder.path))
     }
 
+    func testRemovingAllPackages() throws {
+        try run(with: ["add", "git@github.com:JohnSundell/Files.git"])
+        try run(with: ["add", "git@github.com:JohnSundell/Wrap.git"])
+        try run(with: ["add", "git@github.com:JohnSundell/Unbox.git"])
+
+        let generatedFolder = try folder.subfolder(atPath: "Packages/Generated")
+        let packagesFolder = try generatedFolder.subfolder(named: "Packages")
```
Comment (JohnSundell) [inline Tests/MarathonTests/MarathonTests.swift]: This needs to be updated for Swift 3.1, to use `.build/checkouts` instead.

File: Sources/MarathonCore/PackageManager.swift (hunk 1 above)
Comment (pixyzehn) [inline Sources/MarathonCore/PackageManager.swift]: Yes, I can. I agree with using `@discardableResult`.

File: Tests/MarathonTests/MarathonTests.swift (hunk 4 above)
Comment (pixyzehn) [inline Tests/MarathonTests/MarathonTests.swift]: Sure.

File: Tests/MarathonTests/MarathonTests.swift (hunk 3 above)
Comment (pixyzehn) [inline Tests/MarathonTests/MarathonTests.swift]: This is the strange typo... Thanks!

**Comment by pixyzehn** [comment]
I fixed up all. Thank you for pointing out:)

**Pull request merged**